### PR TITLE
Form 21-8940: Ability to populate Impact on military duty interview questions#14537

### PR DESCRIPTION
**Pull request by zurbergram**
## Description

Form 21-8940: Ability to populate Impact on military duty interview questions

As a Veteran submitting a claim for Individual Unemployability on the Vets.gov platform, I need the ability to populate the military duty interview, so that I can provide additional evidence associated to my Individual Unemployability while completing the Form 21-8940 application interview.

## Testing done
- [x] local testing
- [x] e2e tests

## Screenshots
screen shot 2018-12-07 at 3 32 04 pm
screen shot 2018-12-07 at 3 32 16 pm

## Acceptance criteria
- [x] The user is able to populate the following: 
- [x] If you’re currently serving in the Reserves or the National Guard, does your service-connected disability prevent you from performing your military duties? { radio }
Yes
No
I’m not serving in the Reserves or the National Guard.
- [x] The user is able to move to the next step in the process, Story#14538: Form 21-8940: Ability to populate Impact on recent job applications questions
- [x] Field validations (if applicable per the linked document for Validations above) are enforced in the UI
- [x] Error messages (if applicable per the linked document for Validations above) are displayed in the UI

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/militaryDutyImpact.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  unemployability: {
+    disabilityPreventMilitaryDuties: {
+      'ui:title':
+        'If you’re currently serving in the Reserves or the National Guard, does your service-connected disability prevent you from performing your military duties?',
+      'ui:widget': 'radio',
+      'ui:options': {
+        labels: {
+          yes: 'Yes',
+          no: 'No',
+          alsoNo: 'I’m not serving in the Reserves or the National Guard.',
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/militaryDutyImpact.js]: 😂 Maybe we can put something like `reservesNo` ?

File: src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+} from '../../../../../platform/testing/unit/schemaform-utils';
+import formConfig from '../../config/form';
+import { ERR_MSG_CSS_CLASS } from '../../constants';
+
+describe('Recent Job Applications', () => {
+  const page = formConfig.chapters.disabilities.pages.militaryDutyImpact;
+  const { schema, uiSchema } = page;
+
+  it('should render', () => {
+    const form = mount(
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx]: going to need to update the tests to add `form.unmount()` at the very end due to new linter issues.

File: src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        uiSchema={uiSchema}
+      />,
+    );
+
+    expect(form.find('input').length).to.equal(3);
+  });
+
+  it('should select alsoNo', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx]: this test too `form.unmount()`

File: src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    form.find('form').simulate('submit');
+
+    expect(form.find(ERR_MSG_CSS_CLASS).length).to.equal(0);
+    expect(onSubmit.called).to.be.true;
+  });
+
+  it('should not allow submission with no selection', () => {
+    const onSubmit = sinon.spy();
+    const form = mount(
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx]: this test too `form.unmount()`

**Review by aaronjlech — CHANGES_REQUESTED** [review]
Looks good, just a few suggestions/new linter changes

**Review by aaronjlech — APPROVED** [review]
lgtm

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group - Could I get a code review on this at your earliest convenience? Thanks!

**Review by rianfowler — APPROVED** [review]

**Comment by rianfowler** [comment]
Failing unit test: 
```
Error: Method “simulate” is only meant to be run on a single node. 0 found instead.
at ReactWrapper.single (node_modules/enzyme/build/ReactWrapper.js:1532:17)
at ReactWrapper.simulate (node_modules/enzyme/build/ReactWrapper.js:867:14)
at selectRadio (src/platform/testing/unit/schemaform-utils.jsx:308:6)
at Context.<anonymous> (src/applications/disability-benefits/all-claims/tests/pages/militaryDutyImpact.unit.spec.jsx:41:5)
```

**Pull request merged**